config/environment.rb

```diff
@@ -1,3 +1,6 @@
+# Strip whitespace from environment values for proper API to app handshake.
+ENV.each { |k, v| ENV[k] = v.strip }
```

Comment:
The risk here is we'll mutate all environment variables rather than just the ones our application cares about. This could have a surprise side effect if some rails variable needs leading or trailing whitespace.

What do you think of an AppEnv singleton class that removes the white space for us?
`AppEnv['GITHUB_WEBHOOK_SECRET']` returns `ENV['GITHUB_WEBHOOK_SECRET'].strip`

Reply:
I like that idea - looks like it allows for more granular control over specific environment variables (beyond just .strip as well). Will work on that now.

Reply:
@cbarber is there an example in this project where an environment variable has trailing whitespaces that are needed? If so, why is that the case?

Reply:
I don't think there are examples in **any** project. Wanting to keep leading/trailing whitespace is most likely a non-issue. Just impolite, maybe, to edit env variables for the whole system. Adding an abstraction on top of ENV to strip whitespace adds extra complexity to a solution that is generally just "update your env variables to the correct value". 

The real problem here, I think, is that rails fails to load, or fails to provide some critical function, because env variables are not set correctly and doesn't provide helpful information to the operator that an action is required. Stripping whitespace that was accidentally copy and pasted in solves only a symptom of this issue.